tests/pytorch/test_pytorch_autolog.py

```diff
@@ -457,3 +457,268 @@ def test_autolog_registering_model():
+    Version(pl.__version__) < Version("1.6.0"),
+    reason="`Automatic model checkpointing doesn't exist in pytorch-lightning < 1.6.0",
+)
+def test_automatic_checkpoint_per_30_steps_save_best_only_callback():
+    mlflow.pytorch.autolog(
+        checkpoint=True,
+        checkpoint_monitor="loss_forked_step",
+        checkpoint_mode="min",
+        checkpoint_save_best_only=True,
+        checkpoint_save_weights_only=False,
+        checkpoint_save_freq=30,
+    )
+
+    model = IrisClassification()
+    dm = IrisDataModule()
+    dm.setup(stage="fit")
+    trainer = pl.Trainer(max_epochs=1)
+
+    with mlflow.start_run() as run:
+        trainer.fit(model, dm)
+
+    run_id = run.info.run_id
+
+    metric_keys = {"epoch", "loss_forked_step", "loss", "global_step", "loss_forked"}
+    logged_metrics = mlflow.artifacts.load_dict(
+        f"runs:/{run_id}/checkpoints/latest_checkpoint_metrics.json"
+    )
+    assert logged_metrics["global_step"] == 30
+    assert metric_keys == set(logged_metrics.keys())
+
+    IrisClassification.load_from_checkpoint(
+        mlflow.artifacts.download_artifacts(
+            run_id=run_id, artifact_path="checkpoints/latest_checkpoint.pth"
+        )
+    )
+
+
+@pytest.mark.skipif(
+    Version(pl.__version__) < Version("1.6.0"),
+    reason="`Automatic model checkpointing doesn't exist in pytorch-lightning < 1.6.0",
+)
+def test_automatic_checkpoint_per_epoch_save_best_only_min_monitor_callback():
+    mlflow.pytorch.autolog(
+        checkpoint=True,
+        checkpoint_monitor="val_loss",
+        checkpoint_mode="min",
+        checkpoint_save_best_only=True,
+        checkpoint_save_weights_only=False,
+        checkpoint_save_freq="epoch",
+    )
+
+    model = IrisClassification()
+    dm = IrisDataModule()
+    dm.setup(stage="fit")
+    trainer = pl.Trainer(max_epochs=1)
+
+    with mlflow.start_run() as run:
+        trainer.fit(model, dm)
+
+    run_id = run.info.run_id
+
+    logged_metrics = {k: float(v) for k, v in trainer.callback_metrics.items()}
+    logged_metrics.update({"epoch": 0, "global_step": 33})
+    assert logged_metrics == mlflow.artifacts.load_dict(
+        f"runs:/{run_id}/checkpoints/latest_checkpoint_metrics.json"
+    )
+
+    IrisClassification.load_from_checkpoint(
+        mlflow.artifacts.download_artifacts(
+            run_id=run_id, artifact_path="checkpoints/latest_checkpoint.pth"
+        )
+    )
+
+    callback = [c for c in trainer.callbacks if isinstance(c, MLflowModelCheckpointCallback)][0]
+    trainer.fit_loop.epoch_progress.current.completed += 1
+    trainer._logger_connector._callback_metrics["val_loss"] += 0.1
+    callback.on_train_epoch_end(trainer, model)
+    assert (
+        mlflow.artifacts.load_dict(f"runs:/{run_id}/checkpoints/latest_checkpoint_metrics.json")[
+            "epoch"
+        ]
+        == 0
+    )
+
+    trainer.fit_loop.epoch_progress.current.completed += 1
+    trainer._logger_connector._callback_metrics["val_loss"] -= 0.2
```

Comment:
why do we need to set this?

Reply:
For testing "save_best_only" related logic. e.g. "mode" is "min" , then only loss decreases, the new checkpoint is a better one

Reply:
I feel like this will be broken in the future once the pytorch-lighting team removes `_callback_metrics` or renames it.

Reply:
Updated! :)

Reply:
Updated again.